platform/engineering/design-docs/2020-04-09-separate-content-build.md

```diff
@@ -0,0 +1,498 @@
+assets in the `/applications/` directory.
+
+![Coordinating deployments to S3](images/separated-content-build/coordinating-deployments-to-s3.png)
+
+To ensure that one build won't override files in the other, we'll have to
+1. Update
+   [`vets-website-deploy.sh`](https://github.com/department-of-veterans-affairs/devops/blob/2c63e6d62fa5ad2b6b87478e30d0a7bc2820c252/ansible/deployment/roles/deploy-vets-website/files/vets-website-deploy.sh)
+   to copy only to the `/applications/` directory
+1. Create a new deploy script for the content deployment which copies everything
+   over _except_ the `/applications/` directory
+
+#### Creating temporary static pages for application testing
+- The Webpack configuration will use
+  [`html-webpack-plugin`](https://github.com/jantimon/html-webpack-plugin) to
+  generate temporary landing pages for React applications
+  - This will happen for all build types
+    - Engineers will be able to access the site on a `localhost` build
+    - CI can run E2E tests against production code without the output of the
+      content build
+- The E2E test script in CI will start a process to serve these pages so the
+  tests can run a browser and access the applications
+- **Important note:** These HTML files will not be served in production; the
+  landing pages for the applications come from the content build
+  - To ensure these pages aren't made in production, the deploy script will copy
+    only the files in `/applications/`
+
+#### Miscellaneous
+- Separating the content and application builds will require a standalone content validation job that runs outside of the CI pipeline- probably once a day on a schedule
+  - **Question:** How will we validate links to `vets-website` assets?
+  - **Question:** Where will this job be run? Jenkins? Nomad? Circle?
+- The application build will be responsible for creating the `settings.js`
+  - It's currently a step in the Metalsmith script
+- The landing pages to the applications will be created in the CMS build
+- The CMS repo will no longer import `vets-website`, but rather the new content
+  build repo
+  
+#### React application landing pages
+JavaScript application landing pages will be created during the content build.
+
+**Phase 1: Stopgap solution**  
+To mirror the current process for creating application landing pages, we will
+modify the `createReactPages` Metalsmith plugin to use a JSON file with the
+parameters instead of pulling the properties from the various `manifest.json`
+files. This JSON file will include:
+- URL of the page
+- Application name
+- Whether the file should be made available in production
+
+When we do this, we will update the `manifest.json` documentation and remove
+unused properties from all the files.
+
+**Phase 2: Long-term solution**  
+All application landing pages will be created in the CMS. The page's published
+state will determine whether the application should be made available in
+production.
```

Comment:
Does Flipper play into this decision point at all?

Reply:
It doesn't right now. We may change this later, but :man_shrugging: